addons/mail/models/discuss/discuss_channel.py

```diff
@@ -710,13 +710,25 @@ def _message_compute_parent_id(self, parent_id):
              ('res_id', '=', self.id)
             ]).id
 
+    def _get_allowed_message_post_params(self):
+        params = super()._get_allowed_message_post_params()
+        params.add("special_mentions")
+        return params
+
     @api.returns('mail.message', lambda value: value.id)
     def message_post(self, *, message_type='notification', **kwargs):
         if (not self.env.user or self.env.user._is_public()) and self.is_member:
             # sudo: discuss.channel - guests don't have access for creating mail.message
             self = self.sudo()
         # sudo: discuss.channel - write to discuss.channel is not accessible for most users
         self.sudo().last_interest_dt = fields.Datetime.now()
+        if "everyone" in kwargs.pop("special_mentions", []):
+            kwargs["partner_ids"] = [
+                channel_member.partner_id.id
+                for channel_member in self.channel_member_ids
+                if channel_member.partner_id
+            ]
```

Comment:
```suggestion
            kwargs["partner_ids"] = self.channel_member_ids.partner_id.ids
```

Reply:
Also why do we override existing `partner_ids` rather than appending?
It's unlikely that someone would be missed by `@everyone`, but it would still be better code.